app/src/main/java/uk/ac/ebi/atlas/experimentpage/cellplot/JsonCellPlotController.java

```diff
@@ -0,0 +1,110 @@
+package uk.ac.ebi.atlas.experimentpage.cellplot;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import uk.ac.ebi.atlas.controllers.JsonExceptionHandlingController;
+
+import java.util.Map;
+
+import static uk.ac.ebi.atlas.experimentpage.cellplot.CellPlotType.DEFAULT_PLOT_METHOD;
+
+@RestController
+@RequestMapping(value="/json/cell-plots/{experimentAccession}", method= RequestMethod.GET)
+public class
+JsonCellPlotController extends JsonExceptionHandlingController {
+    private final CellPlotJsonSerializer cellPlotJsonSerializer;
+
+    public JsonCellPlotController(CellPlotJsonSerializer cellPlotJsonSerializer) {
+        this.cellPlotJsonSerializer = cellPlotJsonSerializer;
+    }
+
+    // Our only assumption is that plot parameters are all integers, but we can further generalise if needed
+    private ImmutableMap<String, Integer> parsePlotParametersFromRequestParameters(String plotMethod,
+                                                                                   Map<String, String> requestParams) {
+        var requiredParameters =
+                CellPlotType
+                        .getParameters(plotMethod)
+                        .orElseThrow(() -> new IllegalArgumentException("Unknown plot type " + plotMethod));
+
+        // Check that no param is missing
+        var requiredParametersBuilder = ImmutableMap.<String, Integer>builder();
+        for (var requiredParameter : requiredParameters) {
+            if (!requestParams.containsKey(requiredParameter)) {
+                throw new IllegalArgumentException("Missing parameter " + requiredParameter);
+            } else {
+                try {
+                    requiredParametersBuilder.put(
+                            requiredParameter, Integer.parseInt(requestParams.get(requiredParameter)));
```

Comment:
I had a feeling by looking at these validations that would it be possible to move these validations to any helper class of that package. We can ignore this for the time being and pick this one in the future clean code improvements :)

Reply:
It makes sense. I restrained from disgreagating this into another class because the controller is the only place where this logic is used. If at some point other classes need to use this piece of code we might as well put it in a separate class. This also helps to test it since the logic is non-trivial.